core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -722,23 +725,26 @@ private void syncExistingInodeMetadata(LockedInodePath inodePath, boolean skipLo
 
     if (syncChildren) {
       // Iterate over Alluxio children and process persisted children.
-      mInodeStore.getChildren(inode.asDirectory()).forEach(childInode -> {
-        // If we are only loading non-existing metadata, then don't process any child which
-        // was already in the tree, unless it is a directory, in which case, we might need to load
-        // its children.
-        if (mLoadOnly && inodeChildren.containsKey(childInode.getName()) && childInode.isFile()) {
-          return;
-        }
-        // If we're performing a recursive sync, add each child of our current Inode to the queue
-        AlluxioURI child = inodePath.getUri().joinUnsafe(childInode.getName());
-        mPendingPaths.add(child);
-        // Update a global counter for all sync streams
-        DefaultFileSystemMaster.Metrics.INODE_SYNC_STREAM_PENDING_PATHS_TOTAL.inc();
-        // This asynchronously schedules a job to pre-fetch the statuses into the cache.
-        if (childInode.isDirectory() && mDescendantType == DescendantType.ALL) {
-          mStatusCache.prefetchChildren(child, mMountTable);
-        }
-      });
+      try (CloseableIterator<? extends Inode> children
```

Comment:
Most these changes are just switching to a closable iterator. This way the RocksDB iterators can be consumed during traversal, instead of collecting it at the beginning into a list, then traversing them.

Reply:
I think switching to the `CloseableIterator` should be picked into a standalone separate PR, so that we can better cherry-pick that into branches without adding the partial listing feature as a side effect. Could you keep only the partial listing feature in this PR? So you will have 2 neat changes instead of just 1 :)